### PR TITLE
[bitnami/mariadb] prometheusrule example undefined  $labels

### LINKED ISSUE #10164: mariadb prometheusRule template failed
### Name and Version

bitnami/mariadb 11.0.2

### What steps will reproduce the bug?

1. `helm pull bitnami/mariadb`
2. `tar -xf mariadb-11.0.2.tgz`
3. update values(use code exegesis): 
```yaml
   metrics.prometheusRule.enabled: true
   metrics.enabled: true
   metrics.prometheusRule.rules:
      - alert: MariaDB-Down
      expr: absent(up{job="mariadb"} == 1)
      for: 5m
      labels:
        severity: warning
        service: mariadb
      annotations:
        message: 'MariaDB instance {{ $labels.instance }} is down'
        summary: MariaDB instance is down
```
image
image
image

5. helm template and failed
```
helm template mariadb mariadb -n coding --output-dir  target
```
result
```
Error: template: mariadb/templates/prometheusrules.yaml:23:15: executing "mariadb/templates/prometheusrules.yaml" at <include "common.tplvalues.render" (dict "value" .Values.metrics.prometheusRule.rules "context" $)>: error calling include: template: mariadb/charts/common/templates/_tplvalues.tpl:11:12: executing "common.tplvalues.render" at <tpl (.value | toYaml) .context>: error calling tpl: error during tpl function execution for "- alert: MariaDB-Down\n  annotations:\n    message: MariaDB instance {{ $labels.instance }} is down\n    summary: MariaDB instance is down\n  expr: absent(up{job=\"mariadb\"} == 1)\n  for: 5m\n  labels:\n    service: mariadb\n    severity: warning": parse error at (mariadb/templates/prometheusrules.yaml:3): undefined variable "$labels"

Use --debug flag to render out invalid YAML
```
image

### Are you using any custom parameters or values?
```yaml
metrics:
  enabled: true
  prometheusRule:
    enabled: true
    rules: 
    - alert: MariaDB-Down
      expr: absent(up{job="mariadb"} == 1)
      for: 5m
      labels:
        severity: warning
        service: mariadb
      annotations:
        message: 'MariaDB instance {{ $labels.instance }} is down'
        summary: MariaDB instance is down
```
### What is the expected behavior?

`helm template mariadb mariadb -n coding --output-dir  target` is ok, and have a prometheus rule yaml file

### What do you see instead?
```
Error: template: mariadb/templates/prometheusrules.yaml:23:15: executing "mariadb/templates/prometheusrules.yaml" at <include "common.tplvalues.render" (dict "value" .Values.metrics.prometheusRule.rules "context" $)>: error calling include: template: mariadb/charts/common/templates/_tplvalues.tpl:11:12: executing "common.tplvalues.render" at <tpl (.value | toYaml) .context>: error calling tpl: error during tpl function execution for "- alert: MariaDB-Down\n  annotations:\n    message: MariaDB instance {{ $labels.instance }} is down\n    summary: MariaDB instance is down\n  expr: absent(up{job=\"mariadb\"} == 1)\n  for: 5m\n  labels:\n    service: mariadb\n    severity: warning": parse error at (mariadb/templates/prometheusrules.yaml:3): undefined variable "$labels"
```

### PR DESCRIPTION
**Pull request by lusson-luo**
### Description of the change

before: mariadb use prometheusrules on the annotations is failed.
i do: 
1. update prometheusrules on the annotations use {{ `{{` }} $labels.instance {{ `}}` to {{ $labels.instance }};
2. default prometheusRule.additionalLabels is {}, add is {} judge

### Benefits

use prometheusrules on the annotations is ok

### Possible drawbacks

no drawbacks

### Applicable issues

  - fixes #10164

### Additional information
no

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for creating this PR. Please, note it is needed to sign the commits. Your signature certifies that you wrote the patch or otherwise have the right to contribute the material. You can find more info about this topic, in [the “Sign your work” section from the contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

**Comment by lusson-luo** [comment]
> ok，i had Sign MY Work

**Review by alemorcuq — COMMENTED** [review]
Thanks for the PR. Please, check my comment below.

File: bitnami/mariadb/values.yaml [hunk 1]
```diff
@@ -1171,7 +1171,7 @@ metrics:
     ##      severity: warning
     ##      service: mariadb
     ##    annotations:
-    ##      message: 'MariaDB instance {{ $labels.instance }} is down'
+    ##      message: 'MariaDB instance {{ `{{` }} $labels.instance {{ `}}` }} is down'
```
[the remark this answers is not in the document]
Comment (lusson-luo) [reply bitnami/mariadb/values.yaml:1174]: copy this annotations to prometheusRule , go tmp throw failed with undefined variable "$labels". cause gp tpl will parse  {{ $labels.instance }}, go tmp shouldn't parse it,need think is a string.
Use {{ `{{` }} {{ `}}` }} meaning {{ $labels.instance }} is a string, not to parsing.

[the remark this answers is not in the document]
Comment (lusson-luo) [reply bitnami/mariadb/values.yaml:1174]: a issue is illustrate the context of the question, you can see more info with #10164

**Review by lusson-luo — COMMENTED** [review]
Signed-off-by: lusson <lusson@foxmail.com>

File: bitnami/mariadb/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (lusson-luo) [reply bitnami/mariadb/values.yaml:1174]: @alemorcuq Can my explanation be understood?

**Pull request merged**